### ISSUE TITLE
[bug]: lnwallet's DustLimitForSize doesn't consider p2tr outputs precisely

### ISSUE THREAD

**Issue by hieblmi** [lightningnetwork/lnd#9358]
Calls to `lnwallet.DustLimitForSize(input.P2TRSize)` generate a dust limit based on `input.GenerateUnknownWitness()` which assumes a maximum witness size of 40 bytes.

Similar to `GenerateP2PKH`, we should provide a function `GenerateP2TR` to account for the exact taproot witness size.

See: https://github.com/lightningnetwork/lnd/blob/bb9c680a48cd1075d793cfc97b85f3676b2812c2/lnwallet/parameters.go#L46

**Comment by ziggie1984** [comment]
Hmm not sure if we want to fix it, because we overestimate just by 8 bytes and currently the BOLT spec also does classify all other Pk_Script with a StackElement > OP_0 as unknown witness: https://github.com/lightning/bolts/blob/master/03-transactions.md#dust-limits

If we go that route we would also need to maybe adjust the input script (witness) assumption which is currently after discountn 107.

So not sure if we wanna go this route.

**Comment by Crypt-iQ** [comment]
`input.P2TRSize` is identical to `input.P2WSHSize` so it doesn't hit the unknown case

**Comment by hieblmi** [comment]
> `input.P2TRSize` is identical to `input.P2WSHSize` so it doesn't hit the unknown case

True, both are 34, maybe a clarifying comment on the `case input.P2WSHSize` will suffice then.
